File: myWork/app/src/main/java/com/shindefirm/shopapp/listeners/RecyclerListClickListener.java
package com.shindefirm.shopapp.listeners;

import android.widget.CompoundButton;

public interface RecyclerListClickListener {
    void onListItemClickListener(int position, Object holder, Object object);

    void onSwitchCheckedChanged(CompoundButton buttonView, boolean isChecked,
                                int position, Object object);


}

File: myWork/app/src/main/java/com/shindefirm/shopapp/listeners/RecyclerListClickListener.kt
package com.shindefirm.shopapp.listeners

import android.widget.CompoundButton

interface RecyclerListClickListener {
    fun onListItemClickListener(position: Int, holder: Any?, `object`: Any?)
    fun onSwitchCheckedChanged(
        buttonView: CompoundButton?, isChecked: Boolean,
        position: Int, `object`: Any?
    )
}